gengokit/httptransport/httptransport.go

```diff
@@ -101,6 +111,24 @@ func NewBinding(i int, meth *deftree.ServiceMethod) *Binding {
 		nField.LowCamelName = LowCamelName(nField.Name)
 
 		nBinding.Fields = append(nBinding.Fields, &nField)
+
+		// Emit warnings for certain cases
+		if nField.IsBaseType == false {
+			wrn := []string{
+				"%s.%s is a custom type '%s', only base types and repeated base ",
```

Comment:
Don't do this thing with a string slice that is `strings.Join`'d at runtime. Just use literal concatenation.

```go
wrn := "%s.%s is a custom type '%s', only base types and repeated base " +
    "types are supported. As a result, the generated HTTP " +
    "transport will fail to compile. Remove non-base types."
```